lua/plenary/job.lua

```diff
@@ -76,15 +76,16 @@ function Job:new(o)
     error(debug.traceback("'command' is required for Job:new"))
   end
 
-  if 1 ~= vim.fn.executable(o.command) then
-    error(debug.traceback(o.command..": Executable not found"))
+  local ok, is_exe = pcall(vim.fn.executable, o.command)
+  if ok and 1 ~= is_exe then
+      error(debug.traceback(o.command..": Executable not found"))
   end
 
   local obj = {}
 
   obj.command = o.command
   obj.args = o.args
-  obj.cwd = o.cwd and vim.fn.expand(o.cwd, true)
+  obj.cwd = o.cwd and uv.fs_realpath(o.cwd)
```

Comment:
Overall we should do this PR so good job on pointing that out.
But `vim.fn.expand ~= uv.fs_realpath`
For example: `~/repos` doesn't expands to `/home/conni/repos` but will throw an error.

@tami5 tried to wrote a lua expand but that is currently buggy and doesn't support windows. See
https://github.com/nvim-lua/plenary.nvim/blob/9588829c150f2bcc09f0e1e7eeadfb9e7b85812c/lua/plenary/path.lua#L190-L212

Maybe we should not do `vim.fn.expand` and just expect a already expanded path. And only do expand
`if not vim.in_fast_event() then vim.fn.expand`

Reply:
Wouldn't `fs_realpath` be better than doing nothing at all? And is it necessary that `cwd` is home path and variable expanded?

If a user really needs it then they can always call `vim.fn.expand` when passing in cwd, but for all other cases some sensible path expansion is pretty convenient.

Reply:
I would still wanna do vim.fn.expand when we are not in tight lua callback. Because i am not sure if we are always expanding upfront. So something like this: `Telescope picker cwd=~/repos` might break. Thats what i am saying otherwise we could do `uv.fs_realpath` i just wanted to point out that its not the same :)